libvips/foreign/jxlload.c

```diff
@@ -0,0 +1,969 @@
+		printf( "JXL_DEC_FRAME\n" );
+		break;
+
+	case JXL_DEC_DC_IMAGE:
+		printf( "JXL_DEC_DC_IMAGE\n" );
+		break;
+
+	case JXL_DEC_FULL_IMAGE:
+		printf( "JXL_DEC_FULL_IMAGE\n" );
+		break;
+
+	case JXL_DEC_JPEG_RECONSTRUCTION:
+		printf( "JXL_DEC_JPEG_RECONSTRUCTION\n" );
+		break;
+
+	default:
+		g_assert_not_reached();
+	}
+}
+
+static void
+vips_foreign_load_jxl_print_info( JxlBasicInfo *info )
+{
+	printf( "JxlBasicInfo:\n" );
+	printf( "    have_container = %d\n", info->have_container );
+	printf( "    xsize = %d\n", info->xsize );
+	printf( "    ysize = %d\n", info->ysize );
+	printf( "    bits_per_sample = %d\n", info->bits_per_sample );
+	printf( "    exponent_bits_per_sample = %d\n", 
+		info->exponent_bits_per_sample );
+	printf( "    intensity_target = %g\n", info->intensity_target );
+	printf( "    min_nits = %g\n", info->min_nits );
+	printf( "    relative_to_max_display = %d\n", 
+		info->relative_to_max_display );
+	printf( "    linear_below = %g\n", info->linear_below );
+	printf( "    uses_original_profile = %d\n", 
+		info->uses_original_profile );
+	printf( "    have_preview = %d\n", info->have_preview );
+	printf( "    have_animation = %d\n", info->have_animation );
+	printf( "    orientation = %d\n", info->orientation );
+	printf( "    num_color_channels = %d\n", info->num_color_channels );
+	printf( "    num_extra_channels = %d\n", info->num_extra_channels );
+	printf( "    alpha_bits = %d\n", info->alpha_bits );
+	printf( "    alpha_exponent_bits = %d\n", info->alpha_exponent_bits );
+	printf( "    alpha_premultiplied = %d\n", info->alpha_premultiplied );
+	printf( "    preview.xsize = %d\n", info->preview.xsize );
+	printf( "    preview.ysize = %d\n", info->preview.ysize );
+	printf( "    animation.tps_numerator = %d\n", 
+		info->animation.tps_numerator );
+	printf( "    animation.tps_denominator = %d\n", 
+		info->animation.tps_denominator );
+	printf( "    animation.num_loops = %d\n", info->animation.num_loops );
+	printf( "    animation.have_timecodes = %d\n", 
+		info->animation.have_timecodes );
+}
+
+static void
+vips_foreign_load_jxl_print_format( JxlPixelFormat *format )
+{
+	printf( "JxlPixelFormat:\n" );
+	printf( "    data_type = " );
+	switch( format->data_type ) {
+	case JXL_TYPE_UINT8: 
+		printf( "JXL_TYPE_UINT8" );
+		break;
+
+	case JXL_TYPE_UINT16: 
+		printf( "JXL_TYPE_UINT16" );
+		break;
+
+	case JXL_TYPE_UINT32: 
+		printf( "JXL_TYPE_UINT32" );
+		break;
+
+	case JXL_TYPE_FLOAT: 
+		printf( "JXL_TYPE_FLOAT" );
+		break;
+
+	default:
+		printf( "(unknown)" );
+		break;
+	}
+	printf( "\n" );
+	printf( "    num_channels = %d\n", format->num_channels );
+	printf( "    endianness = %d\n", format->endianness );
+	printf( "    align = %zd\n", format->align );
+}
+#endif /*DEBUG*/
+
+static JxlDecoderStatus 
+vips_foreign_load_jxl_process( VipsForeignLoadJxl *jxl )
+{
+	JxlDecoderStatus status;
+
+	while( (status = JxlDecoderProcessInput( jxl->decoder )) == 
+		JXL_DEC_NEED_MORE_INPUT ) {
+		size_t unused;
```

Comment:
The name `unused` is a bit ambiguous - perhaps `bytes_remaining` instead?

Reply:
Agreed.